src/text-editor.coffee

```diff
@@ -2977,6 +2972,14 @@ class TextEditor extends Model
 
   logScreenLines: (start, end) -> @displayBuffer.logLines(start, end)
 
+  emitWillInsertTextEvent: (text) ->
+    result = true
+    cancel = -> result = false
+    willInsertEvent = {cancel, text}
+    @emit('will-insert-text', willInsertEvent) if includeDeprecatedAPIs
```

Comment:
I think we can remove a lot of these; core doesn't depend on emissary anymore. I'm not sure when exactly we want to remove them.